internal/handlers/handler_verify.go

```diff
@@ -230,12 +244,53 @@ func updateActivityTimestamp(ctx *middlewares.AutheliaCtx, isBasicAuth bool, use
 		return nil
 	}
 
-	// Mark current activity
+	// Mark current activity.
 	userSession.LastActivity = ctx.Clock.Now().Unix()
 	return ctx.SaveSession(userSession)
 }
 
-// VerifyGet is the handler verifying if a request is allowed to go through
+func verifySessionIsUpToDate(ctx *middlewares.AutheliaCtx, targetURL *url.URL, userSession *session.UserSession) (err error) {
+	// TODO: Add a check for LDAP password changes based on a time format attribute. See https://docs.authelia.com/security/threat-model.html#potential-future-guarantees
+
+	refresh, interval := ctx.Providers.UserProvider.GetRefreshSettings()
+
+	ctx.Logger.Tracef("Checking if we need to update session")
+	if refresh && userSession.Username != "" && targetURL != nil && ctx.Providers.Authorizer.URLHasGroupSubjects(*targetURL) && (interval == 0 || userSession.RefreshTTL.Before(ctx.Clock.Now())) {
+		ctx.Logger.Debugf("Checking for updated groups from the authentication backend for user %s", userSession.Username)
+		details, err := ctx.Providers.UserProvider.GetDetails(userSession.Username)
+		// Only update the session if we could get the new details.
+		if err != nil {
+			return err
+		}
+
+		added, removed := utils.SliceStringDelta(userSession.Groups, details.Groups)
+		if len(added) == 0 && len(removed) == 0 {
+			ctx.Logger.Debugf("No updated groups detected for %s", userSession.Username)
+			if interval == 0 {
+				// Skip updating the session if nothing changed and interval is 0.
+				return nil
+			}
+		} else {
+			var delta []string
+			if len(added) != 0 {
+				delta = append(delta, fmt.Sprintf("Added: %s.", strings.Join(added, ", ")))
+			}
+			if len(removed) != 0 {
+				delta = append(delta, fmt.Sprintf("Removed: %s.", strings.Join(removed, ", ")))
+			}
+			ctx.Logger.Debugf("Updated groups detected for %s. %s", userSession.Username, strings.Join(delta, " "))
+			userSession.Groups = details.Groups
```

Comment:
I think we should also update the emails.

Reply:
Done.